### PR TITLE
Backport to 5.0.x: Issue 5193

### PR DESCRIPTION
**Pull request by jlucovsky**
This PR contains the backport from [5188](https://redmine.openinfosecfoundation.org/issues/5188) to `master-5.0.x`

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [5193](https://redmine.openinfosecfoundation.org/issues/5193)

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by victorjulien** [comment]
again
```
  CC       decode-mpls.o
app-layer-ssl.c:1525:1: error: unused function 'HaveEntireRecord' [-Werror,-Wunused-function]
HaveEntireRecord(const SSLStateConnp *curr_connp, const uint32_t input_len)
^
app-layer-ssl.c:1533:1: error: unused function 'RecordAlreadyProcessed' [-Werror,-Wunused-function]
RecordAlreadyProcessed(const SSLStateConnp *curr_connp)
^
```

File: src/app-layer-ssl.c [hunk 1]
```diff
@@ -1370,10 +1433,195 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
...
+    return 0;
+error:
+    curr_connp->trec_len = 0;
+    curr_connp->trec_pos = 0;
+    ValidateTrecBuffer(curr_connp);
+    return -1;
+}
+
+static inline bool
+HaveEntireRecord(const SSLStateConnp *curr_connp, const uint32_t input_len)
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1525]: remove

File: src/app-layer-ssl.c [hunk 2]
```diff
@@ -1370,10 +1433,195 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
...
+{
+    return (curr_connp->bytes_processed + input_len) >=
+                      (curr_connp->record_length + SSLV3_RECORD_HDR_LEN);
+}
+
+#ifdef DEBUG_VALIDATION
+static inline bool
+RecordAlreadyProcessed(const SSLStateConnp *curr_connp)
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1533]: remove, unused

File: src/app-layer-ssl.c (hunk 1 above)
Comment (victorjulien) [reply src/app-layer-ssl.c:1525]: unused

File: src/app-layer-ssl.c [hunk 3]
```diff
@@ -1370,10 +1433,195 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
...
+}
+
+/**
+ *  \retval parsed number of consumed bytes
+ *  \retval < 0 error
+ */
 static int SSLv3ParseHandshakeType(SSLState *ssl_state, const uint8_t *input,
                                    uint32_t input_len, uint8_t direction)
 {
+#if 0
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1622]: remove

File: src/app-layer-ssl.c [hunk 4]
```diff
@@ -1427,6 +1675,11 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, const uint8_t *input,
                            "direction!");
                 break;
             }
+#if 1
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1678]: remove ifdef and `#else` block

**Comment by jlucovsky** [comment]
Continued in #7281

**Pull request closed without merging**